source/lac/petsc_vector_base.cc

```diff
@@ -49,7 +49,7 @@ namespace PETScWrappers
         }
       // else see if we are dealing
       // with a parallel vector
-      else if (dynamic_cast<const PETScWrappers::MPI::Vector *>(&vector) != nullptr)
```

Comment:
From my understanding, we only want to treat `PETScWrappers::Vector` special (we are doing this above).
Since we need `VectorBase` for the test `vector_warp_01`, I want to extend this to all remaining cases of a `PETScWrappers` vector.